test/test_public_bindings.py

```diff
@@ -307,41 +307,74 @@ def test_module(modname):
+                        why_is_public = f"it is inside the module's ({modname}) `__all__`" if is_all else \
+                            "it is an attribute that does not start with `_` on a module that " \
+                            "does not have `__all__` defined"
+                    else:
+                        assert is_all
+                        why_is_public = f"it is not inside the module's ({modname}) `__all__`"
+
+                    if looks_public:
+                        why_looks_public = "it does look public because it follows the rules from the doc above " \
+                            "(does not start with `_` and has a proper `__module__`)."
+                    else:
+                        why_looks_public = why_not_looks_public
+
+                    failure_list.append(f"# {modname}.{elem}:")
+                    is_public_str = "" if is_public else " NOT"
+                    failure_list.append(f"  - Is{is_public_str} public: {why_is_public}")
+                    looks_public_str = "" if looks_public else " NOT"
+                    failure_list.append(f"  - Does{looks_public_str} look public: {why_looks_public}")
 
             if hasattr(mod, '__all__'):
                 public_api = mod.__all__
                 all_api = dir(mod)
                 for elem in all_api:
-                    check_one_element(elem, modname, mod, is_public=elem in public_api)
+                    check_one_element(elem, modname, mod, is_public=elem in public_api, is_all=True)
 
             else:
                 all_api = dir(mod)
                 for elem in all_api:
                     if not elem.startswith('_'):
-                        check_one_element(elem, modname, mod, is_public=True)
+                        check_one_element(elem, modname, mod, is_public=True, is_all=False)
 
         for _, modname, ispkg in pkgutil.walk_packages(path=torch.__path__, prefix=torch.__name__ + '.'):
             test_module(modname)
 
         test_module('torch')
-        msg = "Following new APIs ( displayed in the form (module, element, element module) )" \
-              " were added that do not meet our guidelines for public API" \
-              " Please review  https://github.com/pytorch/pytorch/wiki/Public-API-definition-and-documentation" \
-              " for more information:\n" + "\n".join(map(str, failure_list))
+
+        msg = "All the APIs below do not meet our guidelines for public API from " \
+              "https://github.com/pytorch/pytorch/wiki/Public-API-definition-and-documentation.\n"
+        msg += "Make sure that everything that is public is expected (in particular that the module " \
+            "has a properly populated `__all__` attribute) and that everything that is supposed to be public " \
+            "does look public (it does not start with `_` and has a `__module__` that is properly populated)."
+        msg += "\n\nFull list:\n"
+        msg += "\n".join(map(str, failure_list))
```

Comment:
Add two more bullet point: "if you want to make it public: do this", same for look public